plugins/gradle/src/org/jetbrains/plugins/gradle/service/settings/IdeaGradleProjectSettingsControlBuilder.java

```diff
@@ -284,22 +292,43 @@ private void deduceGradleHomeIfPossible() {
   private void addGradleComponents(PaintAwarePanel content, int indentLevel) {
     myGradlePanel = addComponentsGroup(GradleConstants.GRADLE_NAME, content, indentLevel, panel -> { //NON-NLS GRADLE_NAME
       addGradleChooserComponents(panel, indentLevel + 1);
-      addGradleJdkComponents(panel, indentLevel + 1);
+      if (GradleDaemonJvmHelper.isProjectUsingDaemonJvmCriteria(myInitialSettings)) {
```

Comment:
The `GradleDaemonJvmHelper.isProjectUsingDaemonJvmCriteria` resolves the Gradle version and properties file on the EDT thread. It may cause the UI freeze in WSL projects. 

Could you please consider to pre-evaluate this information on project startup?
For example, you can create a new Gradle JVM macro or create new fields in the `GradleProjectSettings`. And create a separate function for it.

![image](https://github.com/user-attachments/assets/f7b4e66b-c80a-422b-87d9-cc8484a26e30)